src/components/PopupWithForm.js

```diff
@@ -16,8 +17,17 @@ export default class PopupWithForm extends Popup {
     return data;
   }
 
+  setSubmitText(submit, submitText = "Saving...") {
+    if (submit) {
+      this._popupSubmit.textContent = submitText;
+    } else {
+      this._popupSubmit.textContent = "Save";
+    }
+  }
+
   setEventListeners() {
-    this._popupForm.addEventListener("submit", () => {
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
```

Comment:
Needs correction: you should close the modal only after the successful api response in `then` block in index.js